hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleUniqueToken.java

```diff
@@ -21,40 +21,73 @@
  */
 
 import com.google.common.base.MoreObjects;
+import com.hedera.services.state.merkle.internals.IdentityCodeUtils;
 import com.hedera.services.state.submerkle.EntityId;
 import com.hedera.services.state.submerkle.RichInstant;
 import com.swirlds.common.io.SerializableDataInputStream;
 import com.swirlds.common.io.SerializableDataOutputStream;
 import com.swirlds.common.merkle.utility.AbstractMerkleLeaf;
 
 import java.io.IOException;
+import java.time.Instant;
+import java.util.Arrays;
 import java.util.Objects;
 
 /**
  * Represents an uniqueToken entity. Part of the nft implementation.
  */
 public class MerkleUniqueToken extends AbstractMerkleLeaf {
-	public static final int UPPER_BOUND_METADATA_BYTES = 1024;
+	private static final int TREASURY_OWNER_CODE = 0;
+
 	static final int MERKLE_VERSION = 1;
 	static final long RUNTIME_CONSTRUCTABLE_ID = 0x899641dafcc39164L;
 
-	private EntityId owner;
-	private RichInstant creationTime;
+	public static final int UPPER_BOUND_METADATA_BYTES = 1024;
+
+	private int ownerCode;
+	private int creationNanos;
+	private long creationSecs;
```

Comment:
Here is what I was thinking. For convert the RichInstant into primitive values, overall math will be like this. The RichInstant was created by `UniqueToken` and it will take probably 24 bytes in heap. Here in `MerkleUniqueToken`, the original RichInstant will be an references to the one created by `UniqueToken`, and it shall take 4 bytes (In 64-bit JVM if the memory mode is less than 32 GB) or 8 bytes (64-bit JVM if memory mode is larger than 32 GB), in our case we can suppose the reference will be always 8 bytes. Converting it to primitives, it will become 4 + 8 = 12 bytes. If we wipe out the RichInstant object there or any other places as well, we shall have a good save of memory usage. 
The potential downside (or maybe upside) is if we have multiple objects referencing to the same instance, and one modification of the referenced object will guarantee all referencing objects have the updated value of the referenced object. The consistency will be easily maintained. The de-objectifying changes will lose this feature. Again, I am not sure if this is good or bad.

Reply:
Thanks @ljianghedera , interesting points!

1. The `RichInstant` objects created while handling a `TokenMint` are very small, short-lived objects, and based on our experience so far, the garbage collection of such objects is not a big performance problem.
2. So here we are just targeting the savings in the "old generation" of the heap by using `long` + `int` instead of `RichInstant`.
3. Since `RichInstant` is immutable, there isn't any efficiency lost (or consistency put at risk) by replacing it with primitives.

Reply:
I see the motivation to save the "old generation" of heap space. I was not sure how these objects will be used. Thanks for the explanation. 

If we were not under time pressure, I would suggest we do a comparison of profiling on how much different memory usages are saved by the de-objectifying enhancement.